Fix for Tensorflow 2.1

Zahlii opened the pull request:
See also https://github.com/slundberg/shap/issues/1055

Using this version, I was able to successfully calculate SHAP values for a complicated NN based on TF 2.1.0

Zahlii committed:
fix: fixes for tf 2.1

slundberg commented on shap/explainers/deep/deep_tf.py:
```diff
@@ -146,8 +151,7 @@ def __init__(self, model, data, session=None, learning_phase_flags=None):
 
-        if not tf.executing_eagerly():
-            self._init_between_tensors(self.model_output.op, self.model_inputs)
+        self._init_between_tensors(self.model_output.op, self.model_inputs)
```
Did you try this in eager mode?

FRUEHNI1 replied:
Yes - thats why I included this change. If I do not run this code, I get the following error downstream: 
![image](https://user-images.githubusercontent.com/40830407/75446541-e19eb680-5967-11ea-87c2-a8ea09349716.png)

Zahlii commented:
Hi, while running our internal test suite I still noticed some problems related to shap, will try to fix them and update this MR accordingly.

slundberg committed:
Merge branch 'master' into master

RandallJEllis committed:
Corrected spelling of "deprication"

Corrected spelling of "deprication" and "depricated" to "deprecation" and "deprecated", respectively.

slundberg committed:
Fix #1069

slundberg committed:
Merge pull request #1068 from RandallJEllis/patch-1

Corrected spelling of "deprication"

slundberg committed:
Bump version tag

slundberg committed:
Merge branch 'master' of https://github.com/slundberg/shap

slundberg committed:
Update fairness notebook

slundberg committed:
Merge branch 'master' of https://github.com/slundberg/shap

jameslamb committed:
changed uses of equality to 'is' for builtin singletons

Commit:
Added string categorical features for dependence plot. Issue: https://github.com/slundberg/shap/issues/373

Commit:
Added example of working with string-valued features to notebook

Commit:
Updated html file with documentation for dependence plot.

PrimozGodec committed:
kmeans: support sparse

rightx2 committed:
Update PyTorchDeepExplainer: prevent 'Warning: unrecognized nn.Module: XXXXX' when using nested custom models

slundberg committed:
Merge pull request #1114 from jameslamb/misc/comparisons

changed uses of equality to 'is' for None, True, and False

slundberg committed:
Merge pull request #1121 from azanovivan02/dependence_string_features

Added string categorical features for dependence plot

slundberg committed:
Merge pull request #1135 from PrimozGodec/kmeans-sparse

[FIX] kmeans: support sparse

slundberg committed:
Merge pull request #1141 from rightx2/torch_add_handles_upgrade

Prevent unexpected `Warning: unrecognized nn.Module`

Commit:
code changes to publish shapjs to npm

Commit:
Matching shapjs version with the pip package version

Commit:
Adding the npm publish script

Commit:
Fix https://github.com/slundberg/shap/issues/1145

Commit:
Adding developer docs

Commit:
Adding package.json file, updating README and webpack dev server

floidgilbert committed:
documentation updates for decision_plot

slundberg committed:
Merge pull request #1150 from floidgilbert/decision-plot-docs-fix

documentation updates for decision_plot

timothyrenner committed:
Refactor force plots to decouple from IPython using _repr_html_.

Commit:
Added log scale for summary plot.

slundberg committed:
Merge pull request #1156 from azanovivan02/add_log_scale

Added log scale for summary plot.

Harivallabha committed:
Very minor grammatical changes in KernelExplainer docstring

ChemEngDataSci committed:
add sklearn.ensemble._gb.GradeitnBoostingClassifier to the list

slundberg committed:
Merge pull request #1161 from Harivallabha/master

Very minor grammatical changes in KernelExplainer docstring

slundberg committed:
Merge pull request #1147 from SachinVarghese/issue-1040

Publishing shapjs to npm registry

slundberg committed:
Merge pull request #1162 from ChemEngDataSci/master

add sklearn.ensemble._gb.GradientBoostingClassifier to the list

slundberg committed:
Merge pull request #1154 from timothyrenner/decouple-ipython

Refactor force plots to decouple from IPython using _repr_html_.

Commit:
Revert "Fix https://github.com/slundberg/shap/issues/1145"

This reverts commit 36f33e8

Commit:
Update fix https://github.com/slundberg/shap/issues/1145 following comments

slundberg committed:
Merge pull request #1148 from ehuijzer/master

Fixes #1145: Force_plot function with link=logit & matplotlib=True, displays incorrect results for feature importance

lrjball committed:
Fixes error with XGBoost 1.1

In the new XGBoost (1.1), `.save_raw()` output is slightly different, which means that `TreeExplainer` doesn't work. See issue #1215. 
The build is currently failing for this reason, with all XGBoost tests failing.

The fix here is to just remove the prefix 'binf' from the model buffer, which as been added in the new version (either intentionally or accidentally - I can't find a PR in XGBoost which mentions this change.)

Fixes #1215

slundberg committed:
Merge pull request #1220 from lrjball/patch-1

Fixes error with XGBoost 1.1

dependabot[bot] committed:
Bump websocket-extensions from 0.1.3 to 0.1.4 in /javascript

Bumps [websocket-extensions](https://github.com/faye/websocket-extensions-node) from 0.1.3 to 0.1.4.
- [Release notes](https://github.com/faye/websocket-extensions-node/releases)
- [Changelog](https://github.com/faye/websocket-extensions-node/blob/master/CHANGELOG.md)
- [Commits](https://github.com/faye/websocket-extensions-node/compare/0.1.3...0.1.4)

Signed-off-by: dependabot[bot] <support@github.com>

QuentinAmbard committed:
make shap explainer serializable with spark models

QuentinAmbard committed:
make shap explainer serializable with spark models

nasir-bhanpuri committed:
Update summary plot to support custom colormaps 

Addresses https://github.com/slundberg/shap/issues/62 for summary plots.

slundberg committed:
Merge pull request #1313 from QuentinAmbard/master

Make shap explainer serializable with spark models

slundberg committed:
Merge pull request #1314 from nasir-bhanpuri/patch-2

Update summary plot to support custom colormaps

dependabot[bot] committed:
Bump lodash from 4.17.15 to 4.17.19 in /javascript

Bumps [lodash](https://github.com/lodash/lodash) from 4.17.15 to 4.17.19.
- [Release notes](https://github.com/lodash/lodash/releases)
- [Commits](https://github.com/lodash/lodash/compare/4.17.15...4.17.19)

Signed-off-by: dependabot[bot] <support@github.com>

slundberg committed:
Major refactor to support new API (backward compat retained)

slundberg committed:
Bump version for new API changes

slundberg committed:
Fixes #872 by using tempfile module

slundberg committed:
Merge branch 'master' of https://github.com/slundberg/shap

slundberg committed:
Fix merge conflict issues

slundberg committed:
Add slicer dependency

slundberg committed:
Add numba dependency

slundberg committed:
Add missing links file

slundberg committed:
Add approximate_interactions import

slundberg committed:
Updated bar plot and Tree support for the new API

YotamElor committed:
Enable KernelShap support for string type input

slundberg committed:
Merge pull request #1327 from YotamElor/master

Enable KernelShap support for string type input

slundberg committed:
Merge pull request #1323 from slundberg/dependabot/npm_and_yarn/javascript/lodash-4.17.19

Bump lodash from 4.17.15 to 4.17.19 in /javascript

slundberg committed:
Cleanup and new clustering support in plots.bar

slundberg committed:
Merge branch 'master' of https://github.com/slundberg/shap

slundberg committed:
Fix memory out-of-bounds error with some multi-output tree models

slundberg committed:
Skip pyspark tests when pyspark fails to load

slundberg committed:
Make hclust_ordering use scipy's optimal ordering function

dependabot[bot] committed:
Bump elliptic from 6.5.2 to 6.5.3 in /javascript

Bumps [elliptic](https://github.com/indutny/elliptic) from 6.5.2 to 6.5.3.
- [Release notes](https://github.com/indutny/elliptic/releases)
- [Commits](https://github.com/indutny/elliptic/compare/v6.5.2...v6.5.3)

Signed-off-by: dependabot[bot] <support@github.com>

slundberg committed:
Merge pull request #1339 from slundberg/dependabot/npm_and_yarn/javascript/elliptic-6.5.3

Bump elliptic from 6.5.2 to 6.5.3 in /javascript

slundberg committed:
Stop warning for the tree path version of Tree explainer

slundberg committed:
Added support for custom orders in indexing

tcbegley committed:
Correct round 2 samples calculation

imatiach-msft committed:
fixed minor doc issue

imatiach-msft committed:
add python 3.8 support

slundberg committed:
Merge pull request #1353 from imatiach-msft/ilmat/minor-doc-fix

fixed minor doc issue in notebook

fabsig committed:
add support for GPBoost

fabsig committed:
add test for gpboost

slundberg committed:
New heatmap plot, new text plot, new working NLP explainer

slundberg committed:
Merge branch 'master' of https://github.com/slundberg/shap

slundberg committed:
More new API updates, better text plots, heatmaps, etc.

gracecarrillo committed:
Issue #1355 Added 'shap.utils'  to packages to solve issue

Modified line:
packages=[ 'shap', 'shap.explainers', 'shap.explainers.other', 'shap.explainers._deep',
 'shap.plots.colors', 'shap.plots', 'shap.benchmark', 'shap.maskers', 'shap.utils' ],

slundberg committed:
New SHAP API update

slundberg committed:
Fix tree unit tests

slundberg committed:
Fix typo

slundberg committed:
isinstance fix for str checks

aokeson committed:
Fix install bug

Make it so SHAP and Explainers can see the plots.colors and utils modules

slundberg committed:
Merge pull request #1381 from aokeson/patch-1

Fix install bug

slundberg committed:
Update Shapley tutorial

slundberg committed:
Merge branch 'master' of https://github.com/slundberg/shap

slundberg committed:
Merge branch 'master' into patch-1

slundberg committed:
Merge pull request #1374 from gracecarrillo/patch-1

Issue #1355 Added 'shap.utils'  to packages to solve issue

slundberg committed:
Merge pull request #1360 from fabsig/master

add support for GPBoost

slundberg committed:
Fix tansformers base class check

slundberg committed:
Merge branch 'master' of https://github.com/slundberg/shap

slundberg committed:
turn off unit test for window builds sake

slundberg committed:
Update appveyor.yml

slundberg committed:
Merge pull request #1354 from imatiach-msft/ilmat/python3.8

add python 3.8 wheel for releases

slundberg committed:
Fix #1391 matplotlib issue and add initial output selection support

slundberg committed:
Add new benchmark purturbations class

slundberg committed:
Fix #1399 and add multi-bar plot support

slundberg committed:
New support for cohort creating and plotting.

suragnair committed:
Missing import

slundberg committed:
Merge pull request #1407 from suragnair/patch-1

Missing import

Commit:
fix isolation forest when max_features < 1

slundberg committed:
Fix #1408 and other cleanup

slundberg committed:
Merge branch 'master' of https://github.com/slundberg/shap

slundberg committed:
Merge pull request #1412 from zhanjiezhu/fix_iforest_max_features

fix isolation forest when max_features < 1

slundberg committed:
Fix #677 and #1413

slundberg committed:
Merge branch 'master' of https://github.com/slundberg/shap

dependabot[bot] committed:
Bump http-proxy from 1.18.0 to 1.18.1 in /javascript

Bumps [http-proxy](https://github.com/http-party/node-http-proxy) from 1.18.0 to 1.18.1.
- [Release notes](https://github.com/http-party/node-http-proxy/releases)
- [Changelog](https://github.com/http-party/node-http-proxy/blob/master/CHANGELOG.md)
- [Commits](https://github.com/http-party/node-http-proxy/compare/1.18.0...1.18.1)

Signed-off-by: dependabot[bot] <support@github.com>

anusham1990 committed:
Update _general.py

slundberg committed:
Merge pull request #1426 from anusham1990/master

Update _general.py

lrjball committed:
Updated docs

lrjball committed:
Added rtd config file

lrjball committed:
Added examples from notebooks to the docs

lrjball committed:
added deps

lrjball committed:
Updated readme with docs badge

lrjball committed:
Updated css for tables to avoid sideways scrollbar

lrjball committed:
Updated css for tables to avoid sideways scrollbar

slundberg committed:
Merge pull request #1428 from lrjball/docs_update

Docs update

slundberg committed:
Address #1419 with a better error message

slundberg committed:
Clean up plot options

slundberg committed:
Clean up parition code and comments

slundberg committed:
Pin slicer

slundberg committed:
Add cohorts example

slundberg committed:
Merge branch 'master' of https://github.com/slundberg/shap

slundberg committed:
Fix setup.py typo

slundberg committed:
Update docs logo

slundberg committed:
Update doc style

slundberg committed:
update doc logo size

slundberg committed:
Doc color update

slundberg committed:
Fix clustering function

slundberg committed:
Fix multi-output bugs

slundberg committed:
Allow for missing base values

slundberg committed:
Documentation notebook for the Exact explainer

slundberg committed:
Doc coloring consistency

slundberg committed:
small doc color change

slundberg committed:
Fix multi-class bug

slundberg committed:
Merge pull request #1425 from slundberg/dependabot/npm_and_yarn/javascript/http-proxy-1.18.1

Bump http-proxy from 1.18.0 to 1.18.1 in /javascript

slundberg committed:
Merge pull request #1255 from slundberg/dependabot/npm_and_yarn/javascript/websocket-extensions-0.1.4

Bump websocket-extensions from 0.1.3 to 0.1.4 in /javascript

slundberg committed:
Fix multi-output for exact, numba jit fails, and update bar plot api

slundberg committed:
Merge branch 'master' of https://github.com/slundberg/shap

slundberg committed:
New documentation notebooks for Exact and Permutation explainers

slundberg committed:
Fix clustering slicing for multi-output explanations

slundberg committed:
doc css fix

slundberg committed:
Code cleanup

Commit:
Fixed summation of token attribution for multiclass along the class axis

Commit:
Added multiclass text explanation notebook

Commit:
notebook cleanup

anusham1990 committed:
Merge pull request #1 from slundberg/master

Updating fork

anusham1990 committed:
Update _image.py

anusham1990 committed:
Update _partition.py

anusham1990 committed:
Update _explainer.py

Added output_names paramater while creating Partition explainer 
Get output_indices from row_result and create a list for it 
Collapses the output_indices and use them to get the sliced_labels from output_names 
Passed sliced_labels to the explanation object

anusham1990 committed:
Update _explanation.py

Added a check similar to feature_names for output_names for Alias

slundberg committed:
Merge pull request #1442 from ryserrao/fix-multioutput-viz

Fixed summation of token attribution for multiclass along the class axis

slundberg committed:
Merge pull request #1443 from anusham1990/master

Updates for images multi-class output

slundberg committed:
Merge pull request #1350 from tcbegley/master

Correct samples per feature calculation in SamplingExplainer

tcbegley committed:
Change SamplingExplainer's use of nsamples

Previously nsamples corresponded to the number of model evaluations, of
which there were two per coalition of features. With this change
nsamples instead counts the number of sampled coalitions (meaning there
will be 2 * nsamples model evaluations)

Commit:
Merge branch 'master' into multiclass-text-explanation-notebook

slundberg committed:
Fix #1453 by adding operator support for Explanation objects

slundberg committed:
update notebooks

slundberg committed:
Allow a None masker

slundberg committed:
Merge branch 'master' of https://github.com/slundberg/shap

Commit:
Added demo notebook for multiclass text explanation

Commit:
Pass features names and outputnames as list to Alias

Commit:
update to feature ordering to order feature importance based on non abs value

Commit:
Documentation cleanup

anusham1990 committed:
Merge pull request #2 from slundberg/master

Updating repo

anusham1990 committed:
Update _explainer.py for multi-class output labels

If output_names are not given explicitly, then current code fails because output_names numpy array cannot be indexed if None. 
Hence, adding an if condition to check before indexing.

Commit:
added new examples in notebook to visualize top global contributing tokens towards a class by slicing the explanation object

slundberg committed:
Merge pull request #1458 from anusham1990/master

Update _explainer.py for multi-class output labels

Commit:
notebook cleanup

slundberg committed:
Add support for identity sorting

slundberg committed:
Merge branch 'master' of https://github.com/slundberg/shap

slundberg committed:
Merge pull request #1447 from tcbegley/master

Change SamplingExplainer's use of nsamples

vivekchettiar committed:
Added extended tokenizer support and tests for text masker

Commit:
Merge branch 'master' into multiclass-text-explanation-notebook

Commit:
Modified plots in notebook based on the order parameter added in _bar.py

slundberg committed:
Fix custom masking

slundberg committed:
Merge branch 'master' of https://github.com/slundberg/shap

slundberg committed:
Example custom masker notebook

slundberg committed:
Merge pull request #1455 from ryserrao/multiclass-text-explanation-notebook

Multiclass text explanation notebook

slundberg committed:
Merge pull request #1470 from vivekchettiar/tokenizer_support

Non-Fast Tokenizer Support - Text Masker

slundberg committed:
format custom masker example

slundberg committed:
Merge branch 'master' of https://github.com/slundberg/shap

PrimozGodec committed:
Use pseudoinverse instead of inverse for nonsingluar matrices in linear regression

slundberg committed:
Merge pull request #1481 from PrimozGodec/fix-singluar-matrix

Use pseudo-inverse instead of inverse for non-singluar matrices in linear regression

Zahlii committed:
Merge branch 'master' of https://github.com/Zahlii/shap

Zahlii committed:
feat: some fixes for latest TF

Zahlii commented:
Closing since its now obsolete, will follow-up with a newer PR.
